app/src/main/java/fr/free/nrw/commons/utils/DialogUtil.kt

```diff
@@ -0,0 +1,161 @@
+    ) {
+        createAndShowDialogSafely(
+            activity = activity,
+            title = title,
+            message = message,
+            positiveButtonText = activity.getString(R.string.yes),
+            negativeButtonText = activity.getString(R.string.no),
+            onPositiveBtnClick = onPositiveBtnClick,
+            onNegativeBtnClick = onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String?,
+        negativeButtonText: String?,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?
+    ) {
+        createAndShowDialogSafely(
+            activity = activity,
+            title = title,
+            message = message,
+            positiveButtonText = positiveButtonText,
+            negativeButtonText = negativeButtonText,
+            onPositiveBtnClick = onPositiveBtnClick,
+            onNegativeBtnClick = onNegativeBtnClick
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        onPositiveBtnClick: Runnable?,
+        onNegativeBtnClick: Runnable?,
+        customView: View?,
+        cancelable: Boolean
+    ) {
+        createAndShowDialogSafely(
+            activity = activity,
+            title = title,
+            message = message,
+            positiveButtonText = activity.getString(R.string.yes),
+            negativeButtonText = activity.getString(R.string.no),
+            onPositiveBtnClick = onPositiveBtnClick,
+            onNegativeBtnClick = onNegativeBtnClick,
+            customView = customView,
+            cancelable = cancelable
+        )
+    }
+
+    @JvmStatic
+    fun showAlertDialog(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String?,
+        onPositiveBtnClick: Runnable?,
+        cancelable: Boolean
+    ) {
+        createAndShowDialogSafely(
+            activity = activity,
+            title = title,
+            message = message,
+            positiveButtonText = positiveButtonText,
+            onPositiveBtnClick = onPositiveBtnClick,
+            cancelable = cancelable
+        )
+    }
+
+    /**
+     * show a dialog
+     * @param activity
+     * @param title
+     * @param message
+     * @param positiveButtonText
+     * @param negativeButtonText
+     * @param onPositiveBtnClick
+     * @param onNegativeBtnClick
+     * @param customView
+     * @param cancelable
+     */
+    private fun createAndShowDialogSafely(
+        activity: Activity,
+        title: String,
+        message: String,
+        positiveButtonText: String? = null,
+        negativeButtonText: String? = null,
+        onPositiveBtnClick: Runnable? = null,
+        onNegativeBtnClick: Runnable? = null,
+        customView: View? = null,
+        cancelable: Boolean = false
```

Comment:
the default for this should be true I believe

Reply:
Thanks for pointing it out